EventFilter/CastorRawToDigi/python/CastorRawToDigi_cfi.py

```diff
@@ -28,5 +29,13 @@
     InputLabel = cms.InputTag("rawDataCollector"),
     CastorCtdc = cms.bool(False),
     UseNominalOrbitMessageTime = cms.bool(True),
-    ExpectedOrbitMessageTime = cms.int32(-1)
-)
+    ExpectedOrbitMessageTime = cms.int32(-1),
+    UnpackZDC = cms.bool(True),
+    #myMap = cms.VPSet(
+     #  cms.PSet(
+      ##     object = cms.string('ElectronicsMap'),
+        #   file = cms.FileInPath('/afs/cern.ch/user/j/jgomez2/ZDCTest/QuickFix/CMSSW_7_5_0/src/EventFilter/CastorRawToDigi/data/old_ZDC_emap.txt')
```

Comment:
Hi Slava and Salavat,

I've implemented all the requested changes, and we have code that runs both new and old data sets. However, I still have one issue I can't seem to get around.

When I run an old data file, I end up filling ZDCDataFramesSorted_hcalDigis as expected, and for all intents and purposes, everything runs fine. Yet, despite giving me the proper results, I still get an error message when processing:
"Cannot unpack: empty/invalid L1T raw data (size = 0) for ID 1352. Returning empty collections!"
I'm assuming that this is because the Fed id 1352 does not exist in old datasets. 

Is this an issue that will prevent us from moving forward, or is this acceptable? If it's ok, I can finalize this PR and submit others for 76X and 80X. If not... well, I suppose I'll continue to search for a workaround.

Cheers,
Chris

P.S. Thanks for the tip about reco, I'll keep that in mind.


> On Nov 19, 2015, at 03:31, Slava Krutelyov <notifications@github.com> wrote:
> 
> In EventFilter/CastorRawToDigi/python/CastorRawToDigi_cfi.py:
> 
> > @@ -28,5 +29,13 @@
> >      InputLabel = cms.InputTag("rawDataCollector"),
> >      CastorCtdc = cms.bool(False),
> >      UseNominalOrbitMessageTime = cms.bool(True),
> > -    ExpectedOrbitMessageTime = cms.int32(-1)
> > -)
> > +    ExpectedOrbitMessageTime = cms.int32(-1),
> > +    UnpackZDC = cms.bool(True),
> > +    #myMap = cms.VPSet(
> > +     #  cms.PSet(
> > +      ##     object = cms.string('ElectronicsMap'),
> > +        #   file = cms.FileInPath('/afs/cern.ch/user/j/jgomez2/ZDCTest/QuickFix/CMSSW_7_5_0/src/EventFilter/CastorRawToDigi/data/old_ZDC_emap.txt')
> 
> please cleanup the "myMap" commented out piece.
> at least rename the private directory file name
> 
> —
> Reply to this email directly or view it on GitHub.
>